File: old/dig2_darpa2/dig2/programs/complexity/gulwani_popl09/Fig4_1.c
#include <stdio.h>
#include <assert.h>

int mainQ(int n, int m){
     int x = 0;
     int y = 0;
     int t = 0;
     //int t1 = 0;
     //int t2 = 0;
     while(x < n){
	  if(y < m){
	       y = y + 1;
	       //t1++;
	  }
	  else{
	       x = x + 1;
	       //t2++;
	  }
	  t++;
     }
     //, int t1, int t2
     //%%%traces: int n, int m, int t
     //dig2:n - t <= 0, -t <= 0    nothing useful ??
     //NOTE: should we expect t = something here ? 
     return 0;
}


int main(int argc, char **argv){
     mainQ(atoi(argv[1]), atoi(argv[2]));
     return 0;
}

File: examples/SymInferASE/complexity/gulwani_pldi09/Fig4_1.java
public class Fig4_1 {

     public static void main (String[] args) {
	  mainQ(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
     }

     public static int mainQ(int id, int n){
	  assert (id >= 0);
	  assert (n > id);
	  int tmp = id + 1;
	  int t = 0;

	  while(tmp != id){
	       if (tmp <= n) {
		    tmp = tmp + 1;
	       }else{
		    tmp=0;
	       }
	       t++;
	  }
	  //%%%traces: int id, int n, int t
	  //dig2: n - t + 1 == 0, -id <= 0, id - n <= -1 
	  return 0;
     }
}

